Source/WebCore/rendering/style/StyleCanvasImage.cpp

```diff
@@ -132,13 +132,6 @@ void StyleCanvasImage::canvasResized(CanvasBase& canvasBase)
     }
 }
 
-void StyleCanvasImage::canvasDestroyed(CanvasBase& canvasBase)
-{
-    ASSERT_UNUSED(canvasBase, is<HTMLCanvasElement>(canvasBase));
```

Comment:
Seems unfortunate to drop this implementation, especially the assertions. But also, nulling out m_element here doesn't hurt.

Reply:
Right, I thought the weak ptr (m_element) would be set to null at this point, but although it is in the HTMLCanvasElement dtor while canvasDestroyed is being called and about to be destroyed/nulled, the dtor is not actually finished, so indeed safer to keep this part as-is.